### PR TITLE
Cleanup the monster_desc() refactoring.

### PR DESCRIPTION
**Pull request by backwardsEric**
Tries to address the changes (except for those in monster/monster-describer.cpp) requested in the comments on https://github.com/hengband/hengband/pull/2981 .

### CODE REVIEW

File: src/blue-magic/blue-magic-caster.cpp [hunk 1]
```diff
@@ -93,20 +93,21 @@ static bool cast_blue_hand_doom(PlayerType *player_ptr, bmc_type *bmc_ptr)
...
-static std::pair<bool, std::string> exe_blue_teleport_back(PlayerType *player_ptr)
+/* 効果が抵抗された場合、返される std::optional には値がありません。*/
+static std::optional<std::string> exe_blue_teleport_back(PlayerType *player_ptr)
 {
     MonsterEntity *m_ptr;
     MonsterRaceInfo *r_ptr;
     auto *floor_ptr = player_ptr->current_floor_ptr;
     if ((floor_ptr->grid_array[target_row][target_col].m_idx == 0) || !player_has_los_bold(player_ptr, target_row, target_col) || !projectable(player_ptr, player_ptr->y, player_ptr->x, target_row, target_col)) {
-        return std::make_pair(true, std::string());
+        return std::optional<std::string>();
```
Comment (habu1010) [inline src/blue-magic/blue-magic-caster.cpp:103]: You can use std::nullopt instead, which is clearer.

File: src/blue-magic/blue-magic-caster.cpp [hunk 2]
```diff
@@ -136,12 +137,12 @@ static bool cast_blue_teleport_back(PlayerType *player_ptr)
...
 
-    const auto [resisted, m_name] = exe_blue_teleport_back(player_ptr);
-    if (resisted) {
+    const auto m_name = exe_blue_teleport_back(player_ptr);
+    if (!m_name.has_value()) {
         return true;
     }
 
-    msg_format(_("%sを引き戻した。", "You command %s to return."), m_name.data());
+    msg_format(_("%sを引き戻した。", "You command %s to return."), m_name.value().data());
```
Comment (habu1010) [inline src/blue-magic/blue-magic-caster.cpp:145]: Using the arrow operator, you can shorten it to m_name->data().

File: src/blue-magic/blue-magic-caster.cpp (hunk 1 above)
Comment (backwardsEric) [reply src/blue-magic/blue-magic-caster.cpp]: The forced push should have std::nullopt for the returns that used std::optional<std::string>().

File: src/blue-magic/blue-magic-caster.cpp (hunk 2 above)
Comment (backwardsEric) [reply src/blue-magic/blue-magic-caster.cpp]: That should be done in the forced push.

**Comment by backwardsEric** [comment]
The latest forced push rebased to the current state of the develop branch.

**Review by Hourier — APPROVED** [review]
Thank you for the fixes.
I checked the codes and found no problem.
I approve this.

**Pull request merged**